recipes/msix/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+        self._cmake.definitions["USE_VALIDATION_PARSER"] = self.options.use_validation_parser
+        self._cmake.definitions["XML_PARSER"] = self.options.xml_parser
+        self._cmake.configure()
+        return self._cmake
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+            self.options.crypto_lib = "crypt32"
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, "14")
+
+    def requirements(self):
+        if self.settings.os == "Linux" and not self.options.skip_bundles:
+            self.requires("icu/69.1")
+        if self.options.crypto_lib == "openssl":
+            self.requires("openssl/1.0.2t")
+        if self.options.use_external_zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.xml_parser == "xerces":
+            self.requires("xerces-c/3.2.3")
+
+    def validate(self):
+        if self.settings.os != "Android" and self.options.xml_parser == "javaxml":
+            raise ConanInvalidConfiguration("javaxml is supported only for Android")
+        if self.settings.os == "Linux" and self.settings.compiler != "clang":
+            raise ConanInvalidConfiguration("Only clang is supported on Linux")
+        # openssl/1.0.2t is not available for Macos+armv8 at the moment
+        if self.settings.os == "Macos" and self.options.crypto_lib == "openssl":
+            raise ConanInvalidConfiguration("openssl is not supported for MacOS at the moment")
+        if self.settings.os != "Macos" and self.options.xml_parser == "applexml":
+            raise ConanInvalidConfiguration("applexml is supported only for MacOS")
+        if self.settings.os != "Windows" and self.options.crypto_lib == "crypt32":
+            raise ConanInvalidConfiguration("crypt32 is supported only for Windows")
+        if self.settings.os != "Windows" and self.options.xml_parser == "msxml6":
+            raise ConanInvalidConfiguration("msxml6 is supported only for Windows")
+        if self.options.pack:
+            if self.settings.os == "Macos":
+                if not self.options.use_external_zlib:
+                    raise ConanInvalidConfiguration("Using libCompression APIs and packaging features is not supported")
+                if self.options.xml_parser != "xerces":
+                    raise ConanInvalidConfiguration("Xerces is the only supported parser for MacOS pack")
+            if not self.options.use_validation_parser:
+                raise ConanInvalidConfiguration("Packaging requires validation parser")
+        # the current CCI xerces-c recipe is not supported
+        if (self.options.xml_parser == "xerces" and
+            self.options["xerces-c"].char_type != "char16_t"):
+                raise ConanInvalidConfiguration("Only char16_t is supported for xerces-c")
```

Comment:
CCI doens't provide a binary package for `xerces` and `xerces:char16_t` but if the recipe works and build with that combination of requirements and options, we shouldn't raise here, let people at home that are building from sources use this combination. 

If it is because compile/link errors, then of course, keep this exception, but then the comment is probably not accurate.

```suggestion
```

Reply:
It seems that `char16_t` is only compatible option: https://github.com/microsoft/msix-packaging/blob/v1.7/src/msix/PAL/XML/xerces-c/XmlObject.cpp#L301 . See `getAttribute` definition, it returns pointer to `XMLCh`, and then `std::u16string` is constructed from it.

Reply:
Then I'd just remove the comment, it is not a CCI thing

```suggestion

        if (self.options.xml_parser == "xerces" and
            self.options["xerces-c"].char_type != "char16_t"):
                raise ConanInvalidConfiguration("Only char16_t is supported for xerces-c")
```

Reply:
[Removed](https://github.com/conan-io/conan-center-index/pull/6475/commits/f25ff8fbc0ce14b40c9da7b14d1ecacde6726158) the comment